src/java/org/apache/cassandra/io/sstable/AbstractSSTableSimpleWriter.java

```diff
@@ -81,6 +85,12 @@ protected void addIndexGroup(Index.Group indexGroup)
         this.indexGroups.add(indexGroup);
     }
 
+    public void setCompressionDictionary(ColumnFamilyStore cfs, CompressionDictionary compressionDictionary)
+    {
+        this.owner = cfs;
+        this.compressionDictionary = compressionDictionary;
+    }
```

Comment:
Instead of passing cfs in this method, would the API be cleaner if the `writer` is created with `cfs`?